src/components/heatmap/utils.js

```diff
@@ -205,3 +253,61 @@ export function heatmapToMousePosition(colI, rowI, {
   }
   return [zoomedMouseX, zoomedMouseY];
 }
+
+/**
+ * Convert a mouse coordinate (x, y) to a heatmap color bar coordinate (cell index, track index).
+ * @param {number} mouseX The mouse X of interest.
+ * @param {number} mouseY The mouse Y of interest.
+ * @param {object} param2 An object containing current sizes and scale factors.
+ * @returns {number[]} [cellI, trackI]
+ */
+export function mouseToCellColorPosition(mouseX, mouseY, {
+  axisOffsetTop,
+  axisOffsetLeft,
+  offsetTop,
+  offsetLeft,
+  colorBarSize,
+  numCellColorTracks,
+  transpose,
+  targetX,
+  targetY,
+  scaleFactor,
+  matrixWidth,
+  matrixHeight,
+  numRows,
+  numCols,
+}) {
+  const cellPosition = transpose ? mouseX - offsetLeft : mouseY - offsetTop;
+  const trackPosition = transpose ? mouseY - axisOffsetTop : mouseX - axisOffsetLeft;
+
+  const tracksWidth = numCellColorTracks * colorBarSize;
+
+  // outside of cell color tracks
+  if (cellPosition < 0 || trackPosition < 0 || trackPosition >= tracksWidth) {
+    return [null, null];
+  }
+
+  // Determine the trackI and cellI values based on the current viewState.
+  const trackI = Math.floor(trackPosition / colorBarSize);
+
+  let cellI;
+  if (transpose) {
+    const viewMouseX = mouseX - offsetLeft;
+    const bboxTargetX = targetX * scaleFactor + matrixWidth * scaleFactor / 2;
+    const bboxLeft = bboxTargetX - matrixWidth / 2;
+    const zoomedOffsetLeft = bboxLeft / (matrixWidth * scaleFactor);
+    const zoomedViewMouseX = viewMouseX / (matrixWidth * scaleFactor);
+    const zoomedMouseX = zoomedOffsetLeft + zoomedViewMouseX;
+    cellI = Math.floor(zoomedMouseX * numCols);
+  } else {
+    const viewMouseY = mouseY - axisOffsetTop;
+    const bboxTargetY = targetY * scaleFactor + matrixHeight * scaleFactor / 2;
+    const bboxTop = bboxTargetY - matrixHeight / 2;
+    const zoomedOffsetTop = bboxTop / (matrixHeight * scaleFactor);
+    const zoomedViewMouseY = viewMouseY / (matrixHeight * scaleFactor);
+    const zoomedMouseY = zoomedOffsetTop + zoomedViewMouseY;
+    cellI = Math.floor(zoomedMouseY * numRows);
+  }
+
+  return [cellI, trackI];
```

Comment:
This should be refactorable into a guard clause were you return out of the first `if` and don't need an `else` like
```javascript
if(transpose){
...
    return [cellI, trackI];
}
....
return [cellI, trackI];
```